Clarify how to handle results of disqualified attempts

mirandapedro opened the pull request:
This PR change clarifies how should the results of disqualified attempts be handled.

It comes as a result of [incident log #58](https://www.worldcubeassociation.org/incidents/58), in which a competitor was disqualified at the start of a round before having finished any attempts. In alignment with Regulation 2j1a, the Delegate recorded all of his attempts as DNFs.

However, this artificially increases the number of competitors in a round (see Regulation 9m1-9m3) and misrepresents what happened in practice.

The proposed wording for 2j1a makes it clear that the rest of the remaining attempts should only be recorded as DNFs if the competitor has finished one or more attempts. Otherwise, if the competitor has finished no attempts, as is the case in the aforementioned incident log, no results should be recorded.

Commit:
Clarify how to handle results of disqualifiied attmpts

This change clarifies how should the results of disqualified attempts be handled.

It comes as a result of [incident log #58](https://www.worldcubeassociation.org/incidents/58), in which a competitor was disqualified at the start of a round before having finished any attempts. In alignment with Regulation 2j1a, the Delegate recorded all of his attempts as DNFs.

However, this artificially increases the number of competitors in a round (see Regulation 9m1-9m3) and misrepresents what happened in practice.

The proposed wording for 2j1a makes it clear that the rest of the remaining attempts should only be recorded as DNFs if the competitor has finished one or more attempts. Otherwise, if the competitor has finished no attempts, as is the case in the aforementioned incident log, no results should be recorded.

xsrvmy commented:
I think this leaves one issue open for discussion: what should be recorded if the competitor is disqualified in the middle for their first attempt (eg. cheating)?

mirandapedro commented:
Perhaps the wording on the PR should changed from "finished" to "started" so that, if a competitor has started an attempt and is disqualified in the middle of it, the attempt that already has started and the following attempts get recorded as DNF.


This is in alignment with the definition of Regulation 9f4 of what constitutes a DNF:


`9f4) The result of an attempt is recorded as DNF (Did Not Finish) if the attempt is disqualified or unsolved/unfinished`


What do you think?

xsrvmy commented:
Yeah started makes more sense. (I'm only a regular competitor btw)

Ivan-Brigidano commented on wca-regulations.md:
```diff
@@ -88,7 +88,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
 - 2j) The WCA Delegate may disqualify a competitor from a specific event.
     - 2j1) If a competitor is disqualified from an event for any reason, they are not eligible for any more attempts in the event.
-        - 2j1a) The results of all remaining attempts in the event are recorded as DNF.
+        - 2j1a) If the competitor has already finished one or more attempts in a round after being disqualified, the results of all remaining attempts in the event are recorded as DNF. If the competitor has finished no attempts, no results should be recorded.
```
```suggestion
        - 2j1a) If the competitor has already finished one or more attempts in a round after being disqualified, the results of all remaining attempts in the event are recorded as DNF. If did not start any attempt, no results are recorded.
```

I think this would solve @xsrvmy's concerns.

xsrvmy replied:
The first part of the sentence should also started?

Ivan-Brigidano replied:
I'm sorry but I don't understand what you mean. Do you mean about my proposal or the original proposal (Pedro's one)?

mirandapedro replied:
I think what @xsrvmy means is that in `If the competitor has already finished one or more attempts in a round after being disqualified` (the first part of my initial proposal), "finished" should also be altered to "started". I'll push a commit that consolidates both of these changes.

Commit:
Improve wording on the results of disqualified attempts

This commit improves the wording on the handling of results of attempts when a competitor is disqualified from a specific event.

The wording was changed from "finished" to "started" so that if a competitor is disqualified while in the middle of their first attempt, their results get recorded as DNF, in alignment with the definition of Regulation 9f4 of what constitutes a DNF.

I chose "started at least one attempt" so that we don't get any weird wording from "started one or more attempts in a round".

I changed "after being disqualified" to "before being disqualified", as it is what was originally intended and was incorrectly committed before.

Finally, I changed "has" to "had" because I believe it is a more appropriate verb tense in the context.

mirandapedro commented:
Thank you Iván for your help and @xsrvmy for your contributions! It doesn't matter if you're a regular competitor as long as you're willing to help out and improve the Regulations. Comments like these are very appreciated!

Ivan-Brigidano commented on wca-regulations.md:
```diff
@@ -88,7 +88,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
 - 2j) The WCA Delegate may disqualify a competitor from a specific event.
     - 2j1) If a competitor is disqualified from an event for any reason, they are not eligible for any more attempts in the event.
-        - 2j1a) The results of all remaining attempts in the event are recorded as DNF.
+        - 2j1a) If the competitor had already started at least one attempt in a round before being disqualified, the results of all remaining attempts in the event are recorded as DNF. If the competitor had started no attempts, no results should be recorded.
```
One small question before merging this PR:

> If the competitor had started no attempts, no results _should_ be recorded.

What is the purpose of this "should"? Wouldn't it be enough to phrase is similar to the previous sentence?

> If the competitor had started no attempts, no results _are recorded_.

mirandapedro replied:
You're right! I'll push another commit fixing this.

Commit:
Improve wording of Regulation 2j1a

As per @Ivan-Brigidano 's request, this commit improves the wording of Regulation 2j1a so that it is consistent with the wording of the previous sentence and the rest of the Regulations.

Ivan-Brigidano reviewed (approved):
LGTM!

lgarron commented on wca-regulations.md:
```diff
@@ -88,7 +88,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
 - 2j) The WCA Delegate may disqualify a competitor from a specific event.
     - 2j1) If a competitor is disqualified from an event for any reason, they are not eligible for any more attempts in the event.
-        - 2j1a) The results of all remaining attempts in the event are recorded as DNF.
+        - 2j1a) If the competitor had already started at least one attempt in a round before being disqualified, the results of all remaining attempts in the event are recorded as DNF. If the competitor had started no attempts, no results are recorded.
```
I have two small suggestions:

- We can avoid some passive voice.
- It's possible that the competitor has "started one attempt", but this attempt doesn't count, if they were awared an extra attempt. I *think* we don't allow round results that are all DNS?

```suggestion
        - 2j1a) If the competitor has already started at least one attempt in a round before being disqualified, the results of all remaining attempts in the event are recorded as DNF. If the competitor has started no attempts (or only has DNS results), no results are recorded.
```

mirandapedro replied:
I agree with your first suggestion, but I'm not sure if the second is necessary. If the competitor started one attempt, but was granted an extra for it and was disqualified before starting the extra, his result should be recorded as a DNF and not a DNS. As per the definition of a DNS in Regulation 9f5, the competitor did not decline the attempt, so it should be recorded as a DNF instead of a DNS.

It is true that we don't record results that are all DNS, but inserting this on 2j1a does not make sense, as this Regulation deals specifically with the results of attempts in which a competitor was disqualified.

What do you think?

SAuroux replied:
> If the competitor started one attempt, but was granted an extra for it and was disqualified before starting the extra, his result should be recorded as a DNF and not a DNS. 

I disagree with this. As soon as the competitor is granted the extra attempt, the started attempt is voided, so the competitor goes back to the state of having started no attempt.
The definition of a DNS in Regulation 9f5 is not relevant here, since 2j1a) is already superceeding this no matter what, i.e.  the granted extra attempt is no different from the remaining attempts of the round which would also not be voluntarily declined in case of a disqualification.

But for the exact same logic I also disagree with @lgarron 's second suggestion, as having only DNS results is already part of having started no attempts. In my view, the following is more explicit: 

> [...] If the competitor has started no attempts, i.e. has no results or only has DNS results, no results are recorded.

Completely unrelated, I also just realized that the first sentence as is currently leaves a loophole because _a round_ is not associated to the round the competitor is disqualified for. Therefore the first sentence should be:

> 2j1a) If the competitor has already started at least one attempt in a round before being disqualified **from it**, the results of all remaining attempts in the event are recorded as DNF. [...]

mirandapedro replied:
> since 2j1a) is already superceeding this no matter what, i.e. the granted extra attempt is no different from the remaining attempts of the round which would also not be voluntarily declined in case of a disqualification.

By bringing up Regulation 2j1a, you agree with my point that the extra attempt should be recorded as a DNF instead of a DNS, as "The results of all remaining attempts in the event are recorded as DNF" includes the extra attempt. 

This is a different situation than what happened in the incident log where a competitor was disqualified before the round has even started. It is true that the attempt is voided whan an extra is granted, but going by the PRs original intent to try and represent in the results of a competition the closest possible to what actually happenned, if a competitor:
- Started their first attempt in a round
- Were granted and extra attempt, and 
- Were disqualified from the round before starting the extra attempt granted

They participated in the round in a general sense (even if the attempt they actually started was voided), and their results for the round should be recorded as DNF instead of not having results at all.

What do you think?

> Completely unrelated, I also just realized that the first sentence as is currently leaves a loophole because a round is not associated to the round the competitor is disqualified for. Therefore the first sentence should be:

`2j1a) If the competitor has already started at least one attempt in a round before being disqualified from it, the results of all remaining attempts in the event are recorded as DNF. [...]`

Nice improvement! I'll incorporate it in the next commit once we've agreed on the previous discussion.

SAuroux replied:
> By bringing up Regulation 2j1a, you agree with my point that the extra attempt should be recorded as a DNF instead of a DNS, as <current 2ja1>

No. You realize that 2ja1 is the regulation discussed here? :)
I was only pointing out that the new version of 2ja1 is no matter what going to include scenarios where not voluntarily declined attempts are recorded as DNS, i.e. when a competitor is disqualified before the round has started.

> What do you think?

You have effectively not given any new argument, so I do of course still disagree. For some reasons that I do not comprehend, you keep insisting that the existence of some voided attempt in the temporal past should make a difference, which I am not subscribing to, as this is not in line with any other logic used in the regulations.

Instead, I insist on considering voided attempts as voided. I therefore disagree with your statement _They participated in the round in a general sense_ and I continue to say that it should not matter if a competitor with zero completed attempts in a round has previously received zero, one or a hundred extra attempts in that round. If you drew a state diagram of the competitor in that round, all three diagrams would look exactly the same.

Ivan-Brigidano replied:
As soon as an extra attempt is granted, the original result is discarded. Therefore, if the competitor discard the extras, this should be considered DNS. This [Incident Log](https://www.worldcubeassociation.org/incidents/41) describe this exact situation.

> [...] If the competitor has started no attempts, i.e. has no results or only has DNS results, no results are recorded.

I like this suggestion. Only that I prefer using parenthesis rather than commas.

Ivan-Brigidano commented on wca-regulations.md:
```diff
@@ -88,7 +88,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
 - 2j) The WCA Delegate may disqualify a competitor from a specific event.
     - 2j1) If a competitor is disqualified from an event for any reason, they are not eligible for any more attempts in the event.
-        - 2j1a) The results of all remaining attempts in the event are recorded as DNF.
+        - 2j1a) If the competitor had already started at least one attempt in a round before being disqualified, the results of all remaining attempts in the event are recorded as DNF. If the competitor had started no attempts, no results are recorded.
```
```suggestion
        - 2j1a) If the competitor has already started at least one attempt in a round before being disqualified from it, the results of all remaining attempts in the event are recorded as DNF. If the competitor has started no attempts (i.e. has no results or only has DNS results), no results are recorded.
```

Ivan-Brigidano committed:
Update wca-regulations.md
